src/transactions/feed/queryHelper.ts

```diff
@@ -282,31 +304,45 @@ async function queryTransactionsFeed({
   address,
   localCurrencyCode,
   params,
+  updateTransactionState,
+  setActiveRequests,
+  activeRequests,
 }: {
   address: string | null
   localCurrencyCode: string
   params: Array<{
     networkId: NetworkId
     afterCursor?: string
   }>
-}): Promise<{ [key in NetworkId]?: QueryResponse }> {
-  const results = await Promise.all(
-    params.map(({ networkId, afterCursor }) =>
-      queryChainTransactionsFeed({
+  updateTransactionState: (networkId: NetworkId, data: QueryResponse | null) => void
+  setActiveRequests: (updateFunc: (prevState: ActiveRequests) => ActiveRequests) => void
+  activeRequests: ActiveRequests
+}): Promise<void> {
+  // Launch all network requests without waiting for each to finish before starting the next
+  const requests = params.map(async ({ networkId, afterCursor }) => {
+    // Prevent duplicate requests for the same network
+    if (activeRequests[networkId]) {
+      Logger.info(TAG, `Skipping fetch for ${networkId} as it is already active`)
+      return
```

Comment:
I'm wondering if we could avoid that check and the need to pass `setActiveRequests` / `activeRequests`
Is the problem that `useAsync` can call `queryTransactionsFeed` multiple times, but we effectively only care about the last one?

Reply:
I added this because some slow networks, such as polygon, take longer than the polling interval to return a result, so we can end up with multiple pending requests that are slow to resolve.